hack/validate-blocked-edges.py

```diff
@@ -1,9 +1,15 @@
 #!/usr/bin/env python
 
 import os
-
+import re
 import util
 
+# Risk names must be CamelCase to be assignable to condition.Reason
+# https://github.com/openshift/api/blob/8891815aa476232109dccf6c11b8611d209445d9/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L1519-L1520C3
+# Note that our regex is stricter, we should never need names more complicated than this.
+NAME_RE_RAW = r'[A-Z]([A-Za-z0-9_]*[A-Za-z0-9])?$'
```

Comment:
we can probably simplift to `^[A-Z][A-Za-z0-9_]*$`, which:

* Includes a starting anchor `^` for symmetry with the ending anchor `$`.  We don't strictly need `^`, because `match(...)` is `Matches zero or more characters at the beginning of the string.`, but explicitly including it avoids ambiguity.
* Collapse `([A-Za-z0-9_]*[A-Za-z0-9])?` to the simpler and functionally identical `[A-Za-z0-9_]*`.

Reply:
Good catch! Addressed.